core/src/forwarding_stage.rs

```diff
@@ -89,65 +90,81 @@ impl ForwardAddressGetter {
     }
 }
 
+/// [`ForwardingClientError`] enum represents failure when sending transactions
+/// over the network.
+#[derive(Debug)]
+enum ForwardingClientError {
+    /// Failed to send transaction to the provided host.
+    Failed,
+    /// Failed to send the transaction because no contact information was found
+    /// for any of the next `NUM_LOOKAHEAD_LEADERS` scheduled leaders.
+    LeaderContactMissing,
```

Comment:
Although I introduced this error, I haven't introduced a separate counter in metrics for it. I think it is enough to know how many has been dropped on send.